examples/blogs__vue-vuex-rest/cypress/integration/utils.js

```diff
@@ -0,0 +1,58 @@
+/* global cy */
+export const resetDatabase = () => {
+  // for complex resets can use NPM script command
+  // cy.exec('npm run reset:database')
+
+  // for simple cases, can just overwrite the data file
+  const data = {
+    todos: []
+  }
+  const str = JSON.stringify(data)
+  cy.writeFile('./data.json', str)
+  cy.wait(1000) // gives json-server a chance to reload
```

Comment:
I **really** don't like this arbitrary wait here. It assumes a lot of things - that the underlying OS fired the change event that fast and that the jsonServer picked it up.

I would prefer hooking into something more robust that offers a guarantee. For instance - you could use `cy.request` in a loop polling the `jsonServer` until it returns you the reset data you expect. That would work in 100% of cases and not involve arbitrary waiting.

If the file takes `1100ms` to update, your test will fail because on the next visit the jsonServer will send the wrong data.

Reply:
a lot of assumptions, for sure, but seems to work. That is not the important thing here.

Reply:
But we literally describe it as an anti pattern in our docs - and the chances of it working correctly across all machines and all operating systems under all loads in CI has a high probability of causing a flaky test.

Would prefer showing "the right way" to do things to our users. I know it's not the important part here but if this does not work properly **the tests will fail**. So its still essential.

Reply:
is having a JSON file a common way of having a database on the server? Or resetting it by overwriting the said file? Concentrating on this would only confuse the reader, and that is not the important part of the entire thing at all. I specifically did not want to have a build step, or a real server, preferring simplicity over code.

Also, while I understand the `wait` is anti pattern, I guarantee that sometimes one will use it.